Refactor data fetching API docs (#30615)
vercel/next.js: docs/api-reference/data-fetching/getInitialProps.md

# getInitialProps

> **Recommended: [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) or [`getServerSideProps`](/docs/basic-features/data-fetching.md#getserversideprops-server-side-rendering)** instead of `getInitialProps`. These new data fetching methods allow you to have a granular choice between static generation and server-side rendering.

Reviewer: ```suggestion
**Recommended: [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) or [`getServerSideProps`](/docs/basic-features/data-fetching.md#getserversideprops-server-side-rendering)** instead of `getInitialProps`. These data fetching methods allow you to have a granular choice between static generation and server-side rendering.
```

# getInitialProps

> **Recommended: [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) or [`getServerSideProps`](/docs/basic-features/data-fetching.md#getserversideprops-server-side-rendering)**.
> 
> These new data fetching methods allow you to have a granular choice between static generation and server-side rendering. Learn more on the documentation for [Pages](/docs/basic-features/pages.md) and [Data Fetching](/docs/basic-features/data-fetching/index.md).

Reviewer 2: ```suggestion
**Recommended: [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) or [`getServerSideProps`](/docs/basic-features/data-fetching.md#getserversideprops-server-side-rendering)** instead of `getInitialProps`. These new data fetching methods allow you to have a granular choice between static generation and server-side rendering.
```

I don't know if we need to link to Pages or Data Fetching

Reviewer 3: These links are broken since `basic-features/data-fetching.md` was deleted

>
> If you're using Next.js 9.3 or newer, we recommend that you use `getStaticProps` or `getServerSideProps` instead of `getInitialProps`.
>

> > These new data fetching methods allow you to have a granular choice between static generation and server-side rendering. Learn more on the documentation for [Pages](/docs/basic-features/pages.md) and [Data Fetching](/docs/basic-features/data-fetching/data-fetching.md).

Reviewer 2: Should this just be /index?

Author: There was a reason I did this, but I don't remember now. But good shout
